admission_control/admission_control_service/src/mocks/mod.rs

```diff
@@ -0,0 +1,2 @@
+/// Mocker for the local mempool
+pub mod local_mock_mempool;
```

Comment:
I'm moving the mocking out of the unit tests, this way I can use it for fuzzing as well.
I could hide this under both `#[cfg(any(test, feature="fuzzing"))]` but I've noticed that mocks are not gated in most parts of the codebase, probably so that they can be re-used freely.

Reply:
we should gate behind [cfg] everything that is not usede in release (mocks, helper functions, etc.) if mocks are only used in tests, then they should be gated. if the codebase is not consistent with this approach, then we should fix it.

Reply:
done